### PR TITLE
AG-8556 Change zoom anchor split into x and y

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-8556
https://ag-grid.atlassian.net/browse/AG-8552

Changes `scrollingPivot` to `anchorPoints` with separate anchors for x and y axes.

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/zoom/zoomAxisDragger.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export class ZoomAxisDragger {
             return newZoom.y;
         }
 
-        // Scale the zoom along the given axis, pivoting on the end of the axis
+        // Scale the zoom along the given axis, anchoring on the end of the axis
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/zoom/zoomAxisDragger.ts:60]: Axis zoom is a separate ticket which I'll work on after this to make it obey the anchors.

File: packages/ag-charts-community/src/options/chart/zoomOptions.ts [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
 export type AgZoomAxes = 'x' | 'y' | 'xy';
 export type AgZoomPanKey = 'alt' | 'ctrl' | 'meta' | 'shift';
-export type AgZoomScrollingPivot = 'pointer' | 'start' | 'end';
 
 export interface AgZoomOptions {
+    /** The anchor point about which to zoom into when scrolling, defaults to `end`. */
+    anchorPoints?: {
+        x: AgZoomAnchorPoint;
+        y: AgZoomAnchorPoint;
+    };
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/options/chart/zoomOptions.ts]: Would it be preferable to make this `xAnchorPoint` and `yAnchorPoint` instead? What do we usually do for things like this? And in relation to the existing `minXRatio`/`minYRatio` options.

Comment (alantreadway) [reply packages/ag-charts-community/src/options/chart/zoomOptions.ts]: It maybe worth inverting the nesting, if there are several properties relating to each axis direction?

e.g.:
```
  xAxis: AgZoomAxisOptions,
  
  ....
  
interface AgZoomAxisOptions {
  enabled?: boolean; // Replaces `axes?: AgZoomAxes`?
  minRatio?: number;
  anchorPoint?: AgZoomAchorPoint;
}
```

File: packages/ag-charts-enterprise/src/features/zoom/zoom.ts [hunk 3]
```diff
@@ -86,17 +93,19 @@ export class Zoom extends _ModuleSupport.BaseModuleInstance implements _ModuleSu
...
+            ctx.interactionManager.addListener('drag', (event) => this.onDrag(event), interactionOpts),
+            ctx.interactionManager.addListener('drag-end', () => this.onDragEnd(), interactionOpts),
+            ctx.interactionManager.addListener('wheel', (event) => this.onWheel(event), interactionOpts),
+            ctx.interactionManager.addListener('hover', () => this.onHover(), interactionOpts),
             ctx.chartEventManager.addListener('axis-hover', (event) => this.onAxisHover(event)),
             ctx.layoutService.addListener('layout-complete', (event) => this.onLayoutComplete(event))
         );
 
-        // Add layout listener
+        this.anchorPoints = new ZoomAnchorPoints();
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/zoom/zoom.ts]: Nit: Maybe just initialise this inline with the field declaration to reduce clutter? (Same for `axisDragger` below?)

File: packages/ag-charts-community/src/options/chart/zoomOptions.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/options/chart/zoomOptions.ts]: Maybe something to consider discussing with Rob + David ^

Comment (lsjroberts) [reply packages/ag-charts-community/src/options/chart/zoomOptions.ts]: Chatted Rob and we agreed to go with `anchorPointX` and `anchorPointY` with no nesting on any of the props for now.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**